ZelBack/src/services/appsService.js

```diff
@@ -3279,11 +3306,13 @@ async function verifyAppSpecifications(appSpecifications, height) {
       throw new Error('Flux App container data folder not specified. If no data folder is whished, use /tmp');
     }
 
-    // check repotag if available for download
-    await verifyRepository(appSpecifications.repotag);
+    if (checkDockerAndWhitelist) {
+      // check repotag if available for download
+      await verifyRepository(appSpecifications.repotag);
 
-    // check repository whitelisted
-    await generalService.checkWhitelistedRepository(appSpecifications.repotag);
+      // check repository whitelisted
+      await generalService.checkWhitelistedRepository(appSpecifications.repotag);
```

Comment:
Shouldnt this check be done _BEFORE_ you call `verifyRepository` (which fires Docker Hub API request)?